pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -34,10 +34,10 @@ mod connection_record;
 mod migration_state;
 mod signature;
 
-#[cfg(test)]
+#[cfg(all(test, feature = "std"))]
```

Comment:
We always run tests with at least the `std` feature. I also want to test `get_challenge` in a `no_std` environment since debug might be implemented differently there (e.g. for accounts it simply prints nothing).

To be able to test this without `std` I need to disable all tests that require `std`.